### PR TITLE
Reject add-on submission if its guid is more than 64chars (bug 1202016)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1202016](https://bugzilla.mozilla.org/show_bug.cgi?id=1202016)

### CODE REVIEW

**Comment by kmaglione** [comment]
Er, I guess this is identical to #714 aside from the commit message?

**Comment by magopian** [comment]
Yeah, created a new branch/bug

**Comment by magopian** [comment]
@kmaglione could you please give it another look?

File: apps/files/utils.py [hunk 1]
```diff
@@ -456,14 +456,18 @@ def parse_xpi(xpi, addon=None, check=True):
...
-    if (not addon
-            and Addon.with_unlisted.filter(guid=xpi_info['guid']).exists()
-            or BlacklistedGuid.objects.filter(guid=xpi_info['guid']).exists()):
-        raise forms.ValidationError(_('Duplicate UUID found.'))
+    guid = xpi_info['guid']
+    if not guid:
+        raise forms.ValidationError(_("Could not find an add-on ID."))
+    if len(guid) > 64:
+        raise forms.ValidationError(
+            _("Add-on ID must be 64 characters or shorter."))
```
Comment (mstriemer) [inline apps/files/utils.py]: I think: `Add-on ID must be 64 characters or less.`

**Comment by mstriemer** [comment]
I'd rather see `64 characters or less` rather than `64 characters or shorter` but I guess it doesn't matter. Using `shorter` feels grammatically incorrect (length of string (shorter) vs number of characters).

r+

**Comment by mstriemer** [comment]
Travis merges with master before running the tests which is why the build failed. New build should include the fix on master.

**Pull request merged**